### PR TITLE
Render Innovation SSI-2001 samples after write (#1640)

### LINKED ISSUE #1640: No bouncy sound with Innovation SSI-2001 in Airball (1987)
> If you run Airball with "airball /inv" in Daum with `[innova]` turned on, you get sound whenever the ball bounces but in Staging with sidmodel=auto I get nothing.

 -- Thank to @Python-Exoproject for flagging this.

### PR DESCRIPTION
**Pull request by kcgen**
This moves to a render-on-write approach instead of the prior threaded "keep the buffer filled" approach.

The "keep the buffer filled" approach works when writes are spaced no tighter than the 1ms tick rate (such as in Ultima 6), however Airball can make multiple writes within a tick, so this approach accounts for this.

Fixes #1640.

https://user-images.githubusercontent.com/1557255/166333041-a2d8b23f-6b44-4c4b-9cae-8b0a55b58114.mp4

### CODE REVIEW

File: src/hardware/innovation.cpp [hunk 1]
```diff
@@ -21,10 +21,12 @@
...
 #include "pic.h"
+#include "support.h"
 
-// Innovation Settings
-// -------------------
-constexpr uint16_t SAMPLES_PER_BUFFER = 2048;
+// Constants
+// ---------
+constexpr auto idle_after_ms = 200;
+constexpr auto amplification = 6;
```
Comment (johnnovak) [inline src/hardware/innovation.cpp:29]: 6 what? Shouldn't we use decibels instead?

File: src/hardware/innovation.cpp [hunk 2]
```diff
@@ -152,88 +144,75 @@ void Innovation::Close()
...
-		playable.Enqueue(std::move(buffer));
+	int16_t sample = 0;
+	while (!service->clock(1, &sample))
+		; // cycle until we have a sample
+	if (!sample) {
+		++silent_frames;
+		return 0;
 	}
+	silent_frames = 0;
+	const auto bounded = clamp(sample * amplification, MIN_AUDIO, MAX_AUDIO);
```
Comment (johnnovak) [inline src/hardware/innovation.cpp:180]: Probably not that important and unrelated, but a soft-clipper would be nicer 😄

**Review by johnnovak — APPROVED** [review]
Looks good on the surface, perhaps I'll test this later with Ultima 6 and Windwalker.

File: src/hardware/innovation.cpp (hunk 1 above)
Comment (kcgen) [reply src/hardware/innovation.cpp:29]: Dropped it here; now just doubling the sample value at the point of return and asserting that it's in-range.

**Pull request merged**